### PR TITLE
Factorize the Well dataset list

### PR DESCRIPTION
**Pull request by LTMeyer**
There was a duplicate of the dataset list in the download script and in the `dataset.py` file.

This PR creates a single dataset list in the `utils.py`. It adds a function to check whether a dataset name belongs to the Well. It factorizes the list.

I've checked that the changes don't modify the outcome of the automatically generated `registry.yaml` file.

### CODE REVIEW

**Review by rubenohana — COMMENTED** [review]
Before I approve, did you run the tutorial to check if this didn't break anything? Thanks!

**Comment by LTMeyer** [comment]
> Before I approve, did you run the tutorial to check if this didn't break anything? Thanks!

I've checked the downloading script and data processing of the Well tutorial still work.

**Review by rubenohana — APPROVED** [review]
LGTM then! This should be a new version of the package?

File: the_well/data/utils.py
Comment (francois-rozet) [inline the_well/data/utils.py:57]: Do we still need dummy?

**Comment by francois-rozet** [comment]
LGTM as well! I just don't know if we still need `dummy`.

File: the_well/data/utils.py
Comment (LTMeyer) [reply the_well/data/utils.py:57]: Dummy is still used in the tests and relevant to other projects. Let's keep it for now but it's good idea to remove it in the future.

**Pull request merged**
